### DISCUSSION TITLE
Scan not only top level modules but also  transitive dependencies in requirements.txt file

### DISCUSSION THREAD

**Discussion by gskouras** [aquasecurity/trivy#4727] [Ideas]
### Description

Trivy currently only scans for CVEs in the top-level modules specified in the requirements.txt file. However, having Trivy extend its scanning capabilities to include transitive dependencies would be precious.

### Target

Git Repository

### Scanner

Vulnerability

**Comment by knqyf263** [comment]
It would be a significant effort to mimic the behavior of pip. We did it for maven, but it was really challenging. The easiest workaround is using `pip freeze`.
https://aquasecurity.github.io/trivy/v0.42/docs/scanner/vulnerability/language/python/#pip
